src/osd/PrimaryLogPG.cc

```diff
@@ -2045,15 +2045,6 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
     }
   }
 
-  // check for op with rwordered and rebalance or localize reads
-  if ((m->has_flag(CEPH_OSD_FLAG_BALANCE_READS) || m->has_flag(CEPH_OSD_FLAG_LOCALIZE_READS)) &&
-      op->rwordered()) {
```

Comment:
If we want to preserve more of this check, I think we could just change it to use
`m->has_flag(CEPH_OSD_FLAG_RWORDERED)`
rather than the ```op->rwordered()``` shortcut (which also returns true on all writes).j

But it's probably safer to do this wholesale revert and get in more testing with a re-thought replacement. :)

Reply:
For tentacle, I'd prefer to just revert the behavior to the prior behavior.  In main, we can introduce more testing and do something more nuanced.